lib/board.rb

```diff
@@ -0,0 +1,74 @@
+class Board
+  attr_accessor :cells
+
+  #initiazies board as 9 element array with blank spaces
+  def initialize
+    @cells = Array.new(9," ")
+
+  end
+
+  #displays board in a traditional tic tac toe fashion
+  def display
+    puts " #{@cells[0]} " + "|" + " #{@cells[1]} " + "|" + " #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} " + "|" + " #{@cells[4]} " + "|" + " #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} " + "|" + " #{@cells[7]} " + "|" + " #{@cells[8]} "
+  end
+
+  #takes in input and returns that value of the cell in the board
+  def position(input)
+    player_position = input.to_i - 1
+    @cells[player_position]
+  end
+
+  #updates the board with the player's token at the desired position
+  def update(input, player)
+    @cells[input.to_i - 1] = player.token
+  end
+
+  #checks if the board has any open spaces
+  def full?
+    if @cells.include?(" ")
+      return false
+    end
+    return true
+  end
+
+  #counts how many turns there have been by checking for "X" or "O" in cells
+  def turn_count
+    counter = 0
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```